storage/sql_migrations/2_discoveryservice.up.sql

```diff
@@ -0,0 +1,49 @@
+-- discovery contains the known discovery services and the highest timestamp
+create table discovery_service
+(
+    id                varchar(36) not null primary key,
+    lamport_timestamp integer     not null
+);
+
+-- discovery_presentation contains the presentations of the discovery services
+create table discovery_presentation
+(
+    id                      varchar(36) not null primary key,
+    service_id              varchar(36) not null,
+    lamport_timestamp       integer     not null,
+    credential_subject_id   varchar     not null,
+    presentation_id         varchar     not null,
+    presentation_raw        varchar     not null,
+    presentation_expiration integer     not null,
+    unique (service_id, credential_subject_id),
+    constraint fk_discovery_presentation_service_id foreign key (service_id) references discovery_service (id) on delete cascade
+);
+
+-- discovery_credential is a credential in a presentation of the discovery service.
+-- We could do without the table, but having it allows to have a normalized index for credential properties that appear on every credential.
+-- Then we don't need rows in the properties table for them (having a column for those is faster than having a row in the properties table which needs to be joined).
+create table discovery_credential
+(
+    id                    varchar(36) not null primary key,
+    presentation_id       varchar(36) not null,
```

Comment:
Easy to incorrectly relate to `presentation_id` in the `discovery_presentation` table? (note by Rein)

Reply:
comment added